client/src/components/createCredentialDialog/createCredentialDialog.js

```diff
@@ -0,0 +1,453 @@
+        alertType: 'error',
+        header: 'Error',
+        message: results
+      });
+    } else {
+      Store.dispatch({
+        type: toastNotificationTypes.TOAST_ADD,
+        alertType: 'success',
+        message: (
+          <span>
+            Credential <strong>{results.name}</strong> successfully created.
+          </span>
+        )
+      });
+      Store.dispatch({
+        type: credentialsTypes.CREATE_CREDENTIAL_HIDE
+      });
+      this.props.getCredentials();
+    }
+  }
+
+  save() {
+    let credential = {
+      name: this.state.credentialName,
+      cred_type: this.props.credentialType,
+      username: this.state.username
+    };
+
+    if (this.state.authorizationType === 'sshKey') {
+      credential.ssh_keyfile = this.state.sshKeyFile;
+    } else {
+      credential.password = this.state.password;
+    }
+
+    if (this.props.credentialType === 'network') {
+      credential.become_method = this.state.becomeMethod;
+      credential.become_user = this.state.becomeUser;
+      credential.become_password = this.state.becomePassword;
+    }
+
+    this.props.addCredential(credential, this.addResultsCallback);
+  }
+
+  setAuthType(authType) {
+    this.setState({ authorizationType: authType });
+  }
+
+  validateForm(changeObj) {
+    return this.validateCredentialName(
+      changeObj.credentialName || this.state.credentialName
+    );
+  }
+
+  nameExists(name) {
+    const { credentials } = this.props;
+    return (
+      credentials &&
+      credentials.find(credential => {
+        return credential.name === name;
+      })
+    );
+  }
+
+  validateCredentialName(credentialName) {
+    return (
+      credentialName &&
+      credentialName.length > 0 &&
+      !this.nameExists(credentialName)
+    );
+  }
+
+  updateCredentialName(event) {
+    this.setState({
+      credentialName: event.target.value,
+      validCredentialName: this.validateCredentialName(event.target.value),
+      formValid: this.validateForm({ credentialName: event.target.value })
+    });
+  }
+
+  updateUsername(event) {
+    this.setState({
+      username: event.target.value
+    });
+  }
+
+  updatePassword(event) {
+    this.setState({
+      password: event.target.value
+    });
+  }
+
+  updateSshKeyFile(event) {
+    this.setState({
+      sshKeyFile: event.target.value
+    });
+  }
+
+  setBecomeMethod(method) {
+    this.setState({
+      becomeMethod: method
+    });
+  }
+
+  updateBecomeUser(event) {
+    this.setState({
+      becomeUser: event.target.value
+    });
+  }
+
+  updateBecomePassword(event) {
+    this.setState({
+      becomePassword: event.target.value
+    });
+  }
+
+  renderFormLabel(label) {
+    return (
+      <Grid.Col componentClass={Form.ControlLabel} sm={5}>
+        {label}
+      </Grid.Col>
+    );
+  }
+
+  renderAuthForm() {
+    const { authorizationType, password, sshKeyFile } = this.state;
+
+    if (authorizationType === 'usernamePassword') {
```

Comment:
Optional: possible switch, or removing the "else" on line ```228``` since they're both returning values